hkmc2/shared/src/test/mlscript/codegen/Scoped.mls

```diff
@@ -0,0 +1,277 @@
+:js
+
+
+:sjs
+let x =
+  let y = 1 + 3 * 4
+  y + 4
+//│ JS (unsanitized):
+//│ let x, y, tmp; tmp = 3 * 4; y = 1 + tmp; x = y + 4;
+//│ x = 17
+
+
+:sjs
+fun f() =
+  if true then
+    let a = 4
+    a + 4
+  else
+    let b = 5
+    b + 9
+//│ JS (unsanitized):
+//│ let f;
+//│ f = function f() {
+//│   let scrut, a, b;
+//│   scrut = true;
+//│   if (scrut === true) { a = 4; return a + 4 } else { b = 5; return b + 9 }
+//│ };
+
+
+
+// `let a` is declared in the correct place,
```

Comment:
Again, it's not "the correct place". There are several possible places where it would make sense, and this one is not the best (to be imprived on later).